### PR TITLE
Add toJson methods to render config objects in JS

### PR DESCRIPTION
**Pull request by zzmp**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 93add63f6585c72a2ae3bddfac61706b0b60c164 available at: https://gist.github.com/2a7514654d43574c1160<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7034.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7034.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c334e4ebcb7c1172968e39b334033ade07c56a0c available at: https://gist.github.com/2a7514654d43574c1160<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7034.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7034.dmg

**Comment by OmegaHeron** [comment]
Tested for basic function - compared to current production build this works as expected.  As to its actual intent, there, no apparent way to test/exercise it.  :)

**Comment by zzmp** [comment]
@OmegaHeron if you go into the scripting interface (ctrl+alt+J), you should be able to call `Render.toJSON()` and see all of the render engine configuration settings.

**Comment by OmegaHeron** [comment]
Cool - looking at it now, but, for QA Testing a little bit of what this is for -- its intent and perhaps some test cases would be good.  As a poke around and have fun thing looking at the functions is great.  ;)

**Comment by OmegaHeron** [comment]
Seems to work - nothing crashes, worlds don't implode - so, for what it's worth, it passes smoke test.
`> Render.toJSON()`
`< {"RenderDeferredTask":{"AmbientOcclusion":{"blurDeviation":2.5,"blurRadius":4,"borderingEnabled":true,"ditheringEnabled":true,"edgeSharpness":1,"enabled":false,"falloffBias":0.0099999997764825821,"gpuTime":0,"numSamples":11,"numSpiralTurns":7,"obscuranceLevel":0.5,"radius":0.5,"resolutionLevel":1},"Antialiasing":{"enabled":false},"DebugDeferredBuffer":{"enabled":false,"mode":0,"size":[0,0,0,0]},"DrawOpaqueDeferred":{"maxDrawn":-1,"numDrawn":137},"DrawOverlay3D":{"maxDrawn":-1,"numDrawn":44,"numItems":44},"DrawStatus":{"showDisplay":false,"showNetwork":false},"DrawTransparentDeferred":{"maxDrawn":-1,"numDrawn":0},"FetchOpaque":{"numItems":161},"FetchTransparent":{"numItems":0},"HitEffect":{"enabled":false},"ToneMapping":{"curve":1,"enabled":true,"exposure":0}},"RenderShadowTask":{"FetchShadowMap":{"numItems":0},"enabled":false}}`

**Comment by zzmp** [comment]
OK, I'll make sure to include intent and test cases in my future PRs. Thanks, @OmegaHeron

File: libraries/shared/src/shared/JSONHelpers.cpp [hunk 1]
```diff
@@ -57,3 +61,63 @@ vec3 vec3FromJsonValue(const QJsonValue& v) {
...
+
+    return json;
+}
+
+void qObjectFromJsonValue(const QJsonValue& j, QObject& o) {
+    const QJsonObject object = j.toObject();
+    for (auto it = object.begin(); it != object.end(); it++) {
+        std::string key = it.key().toStdString();
+        if (it.value().isObject()) {
+            QVariant child = o.property(key.c_str());
```
Comment (zzmp) [inline libraries/shared/src/shared/JSONHelpers.cpp]: Looking up children as a property does not work. This is not yet used, so I'm ok fixing it in #7043.

**Pull request merged**